util/dataframe_ops.py

```diff
@@ -273,41 +282,66 @@ def parse_check(tolerance_file_name, input_file_ref, input_file_cur, factor):
     """
     df_tol = parse_probtest_csv(tolerance_file_name, index_col=[0, 1])
 
-    logger.info("applying a factor of %s to the spread", factor)
     df_tol *= factor
 
     df_ref = parse_probtest_csv(input_file_ref, index_col=[0, 1, 2])
     df_cur = parse_probtest_csv(input_file_cur, index_col=[0, 1, 2])
 
-    logger.info(
-        "checking %s against %s using tolerances from %s",
-        input_file_cur,
-        input_file_ref,
-        tolerance_file_name,
-    )
-
     return df_tol, df_ref, df_cur
 
 
 def check_stats_file_with_tolerances(
     tolerance_file_name, input_file_ref, input_file_cur, factor
 ):
 
-    df_tol, df_ref, df_cur = parse_check(
-        tolerance_file_name, input_file_ref, input_file_cur, factor
-    )
+    if "fof" in input_file_ref:
+        type_f = "fof"
+        ds_tol = pd.read_csv(tolerance_file_name, index_col=0)
+        df_tol = ds_tol * factor
+
+        df_ref = parse_probtest_fof(input_file_ref)
+
+        df_cur = parse_probtest_fof(input_file_cur)
+
+    else:
+        type_f = "stats"
```

Comment:
I suggest renaming this variable to file_type for clarity. Additionally, instead of using raw strings, consider using a Python Enum to make the code more robust and self-documenting. This prevents typos and clearly defines the allowed values.
```
class FileType(Enum):
    FOF = "fof"
    STATS = "stats"
 ```